README.md

```diff
@@ -8,4 +8,47 @@ In order to prevent sensitive credentials to be leaked, please follow this instr
 - Install [git-secrets](https://github.com/awslabs/git-secrets) in your computer.
 - Go into the repo root folder.
 - Run `git secrets --register-aws`.
-- Run `git config --local core.hooksPath .githooks/`.
\ No newline at end of file
+- Run `git config --local core.hooksPath .githooks/`.
+
+## Running tests
+
+The easiest way to run the tests locally is to run the test DB in docker.
+
+Ensure you have the following environment variables set on the terminal session you want to run your tests from:
+```
+POSTGRES_PORT=5433
+API_KEY='anything'
+RUNNING_TEST=true
+```
+
+Then, in the HRM root, run
+
+```shell
+docker compose -f ./docker-compose-test.yml up -d
+jest --verbose                                      # tests will run here 
+docker compose -f ./docker-compose-test.yml down    # if you forget this, the DB will not be cleared down & be in a dirty state for the next run, which could make tests unstable
+```
+
+### Debugging SQL in test runs
+
+If you want to debug the queries hitting the DB as part of the tests, you can use 2 terminal sessions.
+
+In a new terminal session run:
+
+```shell
+docker compose -f ./docker-compose-test.yml up # don't use the '-d' flag!
+```
+
+Then in the original (with the environment variables set) session run
+
+```shell
+jest --verbose 
+```
+
+You should see all the SQL queries run by the tests logged out in the docker terminal.
+
+After the run, Ctrl-C the docker compose session and run 
+```shell
+docker compose -f ./docker-compose-test.yml down
+```
+in any terminal window.
```

Comment:
Update on the thread of tests failing: I've tried this and apparently no query is run against the container (!?.

This are the last logs of the container window, after running the tests.

```
db_1  | waiting for server to shut down...2022-02-10 21:47:11.743 UTC [48] LOG:  received fast shutdown request
db_1  | .2022-02-10 21:47:11.745 UTC [48] LOG:  aborting any active transactions
db_1  | 2022-02-10 21:47:11.747 UTC [48] LOG:  background worker "logical replication launcher" (PID 55) exited with exit code 1
db_1  | 2022-02-10 21:47:11.747 UTC [50] LOG:  shutting down
db_1  | 2022-02-10 21:47:11.784 UTC [48] LOG:  database system is shut down
db_1  |  done
db_1  | server stopped
db_1  | 
db_1  | PostgreSQL init process complete; ready for start up.
db_1  | 
db_1  | 2022-02-10 21:47:11.877 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
db_1  | 2022-02-10 21:47:11.877 UTC [1] LOG:  listening on IPv6 address "::", port 5432
db_1  | 2022-02-10 21:47:11.881 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
db_1  | 2022-02-10 21:47:11.915 UTC [85] LOG:  database system was shut down at 2022-02-10 21:47:11 UTC
db_1  | 2022-02-10 21:47:11.921 UTC [1] LOG:  database system is ready to accept connections
```

Reply:
Because of line 26 of `models/index.js,` we use development config in tests too
`const config = configFile[process.env.NODE_ENV] || configFile.development;`

I can see that `port: '5433',` if I log this config while running tests.

Running the container and then doing in a different console
`➜ psql --host=localhost --dbname=hrmdb --username=hrm --port=5433`
connects me to the DB in the container, running a query outputs it in the container window too and I can query against it successfully.

![Screenshot from 2022-02-11 17-34-27](https://user-images.githubusercontent.com/15805319/153666344-70d71307-3550-4d28-96d8-b73f2436a666.png)

![Screenshot from 2022-02-11 17-34-41](https://user-images.githubusercontent.com/15805319/153666347-5789487f-8695-4d0c-a62e-e84c7210e983.png)

If I change `docker-compose-test.yml` like
```
ports:
      - "5432:5432"
``` 
I receive (as expected) an error indicating that this service is already in use (by my local Postgres service)
![Screenshot from 2022-02-11 17-36-10](https://user-images.githubusercontent.com/15805319/153666692-ffcdca50-49c8-4cfe-9c95-023c608a754c.png)

At this point it seems clear to me that my tests running on hrm, even when the port is set to be `5433` in the config, is not hitting the container at all. Will update if I can find something useful.

Reply:
Yea, dump all the output to a text file, just in case.

Out of interest, does it work if you point the tests 5432 and use your local DB non docker DB? (as in does it connect and run queries, the tests might not pass)

I don't suppose you can step through the tests with your debugger to see what line it hangs on? Otherwise we're reduced to putting console.log everywhere

Reply:
> Out of interest, does it work if you point the tests 5432 and use your local DB non docker DB? (as in does it connect and run queries, the tests might not pass)

You mean like the usual flow for tests until this PR? The tests fails exactly the same as I've shared above.

> I don't suppose you can step through the tests with your debugger to see what line it hangs on? Otherwise we're reduced to putting console.log everywhere

Not sure how to run the debugger, I run in the terminal, but I will try leaving a few logs for now!

Reply:
Changed from Node v14 to v12, that fixes the issues because of Sequelize (: 

![Screenshot from 2022-02-16 14-04-14](https://user-images.githubusercontent.com/15805319/154317715-0a7e9ab1-be4c-4e72-8cab-3331f39390d0.png)

Sorry all the issues @stephenhand.